File: DSA/pranavdixit_590014418/Day_17_Question_2.py
def first_occurrence(arr, target):
    low, high = 0, len(arr) - 1
    result = -1
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            result = mid
            high = mid - 1
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1
    return result

print(first_occurrence([1, 2, 2, 2, 3, 4], 2))
print(first_occurrence([5, 7, 7, 8, 8, 8, 10], 8))
print(first_occurrence([1, 1, 1, 2, 3, 3], 1))

File: DSA/Manushree_590017458/day_17_question_2.c
#include <stdio.h>

int firstOccurrence(int arr[], int size, int target) {
    int low = 0, high = size - 1, result = -1;
    while(low <= high) {
        int mid = low + (high - low) / 2;
        if(arr[mid] == target) {
            result = mid;
            high = mid - 1;
        } else if(arr[mid] < target) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return result;
}

int main() {
    int arr[] = {1, 2, 2, 2, 3, 4};
    int target = 2;
    int size = sizeof(arr) / sizeof(arr[0]);
    printf("%d\n", firstOccurrence(arr, size, target));
    return 0;
}
